integrade/tests/aws_utils.py

```diff
@@ -0,0 +1,204 @@
+    The profile must be named and have its credentials stored in ~/.aws/config
+    The images are associated with the profiles in
+    ~/.config/integrade/aws_config in the following manner:
+
+    aws:
+      profiles:
+        name_of_profile1:
+          images:
+            name_of_image:
+              is_rhel: True
+              image_id: ami-123456789
+              other_key: other_value
+        name_of_profile2:
+          images:
+            name_of_image:
+              is_rhel: False
+              image_id: ami-567890234
+              other_key: other_value
+    """
+    client = aws_session(aws_profile).client('ec2')
+    cfg = config.get_aws_config()
+    image_id = cfg['profiles'][aws_profile]['images'][image_name]['image_id']
+    response = client.run_instances(
+        MaxCount=count,
+        MinCount=count,
+        ImageId=image_id,
+        InstanceType=instance_type)
+    instance_ids = []
+    for instance in response.get('Instances', []):
+        instance_ids.append(instance['InstanceId'])
+    with Pool(len(instance_ids)) as p:
+        p.map(
+            wait_until_running, zip(
+                [aws_profile] * len(instance_ids), instance_ids))
+    return instance_ids
+
+
+def run_instances(aws_profile, image_name, count, run_time):
+    """Create instances and run them for a certain amount of time.
+
+    :param aws_profile: (string) Name of profile as defined in config file
+    :param image_name: (string) Name of image as defined in config file
+    :param count: (int) Number of instances to create.
+    :param run_time: (double) Desired time to let instances run (in seconds)
+
+    :returns: (double) Time from when instances were all live until all
+         instances were all terminated (in seconds).
+    """
+    instance_ids = create_instances(aws_profile, image_name, count=count)
+    # now sleep until instances have run for desired amount of time
+    time_all_live = time.time()
+    time.sleep(run_time)
+    with Pool(len(instance_ids)) as p:
+        p.map(
+            terminate_instance, zip(
+                [aws_profile] * len(instance_ids), instance_ids))
+    time_all_terminated = time.time()
+    return time_all_terminated - time_all_live
+
+
+def terminate_all_instances(aws_profile):
+    """Terminate all instances for a given aws account.
+
+    :param aws_profile: (string) Name of profile as defined in config file
+
+    This is useful when you want to make sure there are no running instances
+    in an account. Terminated instances eventually disappear from the list of
+    instances visible in the EC2 console or via describe_instances(), but this
+    is cannot be controlled by the user (happens on the AWS backend).
+    """
+    client = aws_session(aws_profile).client('ec2')
+    instances_to_terminate = []
+    for reservation in client.describe_instances().get('Reservations', []):
+        for instance in reservation.get('Instances'):
+            if instance['State']['Code'] != 48:
```

Comment:
Maybe worth adding a comment stating which is the state code name or create a constant to make easy to read.